### PR TITLE
Fix project.json's in buildtools

### PR DESCRIPTION
**Pull request by ericstj**
The recent version of NuGet broke the BuildTools package since it
started filtering out files named project.json without any means to
disable:
https://github.com/NuGet/NuGet.Client/commit/59433c7bacaae435a2cfe343cd441ea710579304#diff-7d862ed9e52e34d5ee39d6ccbf6f8a7cR665
Workaround this by naming them differently and renaming during
init-tools.

/cc @weshaggard @joshfree

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj [hunk 1]
```diff
@@ -81,6 +81,8 @@
       <NuProjFiles Include="$(PackagesDir)NuProj\$(NuProjVersion)\tools\**\*.*" />
     </ItemGroup>
     <Copy SourceFiles="@(PackageFiles)" DestinationFolder="$(OutputPath)\PackageFiles\%(RecursiveDir)" SkipUnchangedFiles="true" />
+    <!-- nuget strips files with the name 'project.json' so give a suffix that we'll rename in init-tools -->
+    <Copy SourceFiles="@(PackageFiles)" Condition="'%(FileName)%(Extension)' == 'project.json'" DestinationFiles="$(OutputPath)\PackageFiles\%(RecursiveDir)%(FileName)%(Extension).rename" SkipUnchangedFiles="true" />
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/Microsoft.DotNet.Build.Tasks.csproj]: LOL there isn't anyway to have them not do that?

**Comment by weshaggard** [comment]
LGTM

cc @joperezr we should undo this when we remove the tool-runtime as part of your other work.

**Comment by ericstj** [comment]
Had to upgrade the CoreFx version because NuGet was giving me downgrade warnings during init-tools.

**Comment by joperezr** [comment]
> we should undo this when we remove the tool-runtime as part of your other work.

Will do.

**Comment by joshfree** [comment]
Thanks @ericstj for making progress on this despite the nuget regression

**Pull request merged**